src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -1051,57 +1053,47 @@ ImmutableArray<FieldSymbol> collectTupleElementFields(NamedTypeSymbol tuple)
-                        (_lazyUnderlyingDefinitionToMemberMap = ComputeDefinitionToMemberMap());
-                }
-            }
+                    return _lazyUnderlyingDefinitionToMemberMap ??= computeDefinitionToMemberMap();
 
-            private SmallDictionary<Symbol, Symbol> ComputeDefinitionToMemberMap()
-            {
-                var map = new SmallDictionary<Symbol, Symbol>(ReferenceEqualityComparer.Instance);
-                var members = TupleUnderlyingType.GetMembers();
-
-                // Go in reverse because we want members with default name, which precede the ones with
-                // friendly names, to be in the map.
-                for (int i = members.Length - 1; i >= 0; i--)
-                {
-                    var member = members[i];
-                    switch (member.Kind)
+                    SmallDictionary<Symbol, Symbol> computeDefinitionToMemberMap()
                     {
-                        case SymbolKind.Method:
-                        case SymbolKind.Property:
-                        case SymbolKind.NamedType:
-                            map.Add(member.OriginalDefinition, member);
-                            break;
+                        var map = new SmallDictionary<Symbol, Symbol>(ReferenceEqualityComparer.Instance);
+                        var members = TupleUnderlyingType.GetMembers();
 
-                        case SymbolKind.Field:
-                            var tupleUnderlyingField = ((FieldSymbol)member).TupleUnderlyingField;
-                            if (tupleUnderlyingField is object)
-                            {
-                                map[tupleUnderlyingField.OriginalDefinition] = member;
-                            }
-                            break;
-
-                        case SymbolKind.Event:
-                            var underlyingEvent = (EventSymbol)member;
-                            var underlyingAssociatedField = underlyingEvent.AssociatedField;
-                            // The field is not part of the members list
-                            if (underlyingAssociatedField is object)
+                        // Go in reverse because we want members with default name, which precede the ones with
+                        // friendly names, to be in the map.
+                        for (int i = members.Length - 1; i >= 0; i--)
+                        {
+                            var member = members[i];
+                            switch (member.Kind)
                             {
-                                Debug.Assert((object)underlyingAssociatedField.ContainingSymbol == TupleUnderlyingType);
-                                Debug.Assert(TupleUnderlyingType.GetMembers(underlyingAssociatedField.Name).IndexOf(underlyingAssociatedField) < 0);
-                                map.Add(underlyingAssociatedField.OriginalDefinition, new TupleFieldSymbol(TupleUnderlyingType, underlyingAssociatedField, -i - 1));
```

Comment:
>map.Add(underlyingAssociatedField.OriginalDefinition, new TupleFieldSymbol(TupleUnderlyingType, underlyingAssociatedField, -i - 1)); [](start = 32, length = 132)

It is not clear why this is no longer needed. We don't need/want to create an instance of a TupleFieldSymbol, but it feel like we still want an entry in the map if we wanted to have it in the map before this PR.

Reply:
I'll put the logic back for completeness, but the only place we use this is in `TupleFieldSymbol.AssociatedSymbol` (I don't think event backing fields would use it).

---
In reply to: [426810809](https://github.com/dotnet/roslyn/pull/44231#discussion_r426810809) [](ancestors = 426810809)